### PR TITLE
Schema: fix dashboard revision docs

### PR DESCRIPTION
**Pull request by ryantxu**
The dashboard "revision" property should only be used for plugin dashboards.  Ideally we can remove it but we should make sure it is not confused with version, and should not be set in most as-code workflows

### CODE REVIEW

**Review by nmarrs — APPROVED** [review]
LGTM

File: kinds/dashboard/dashboard_kind.cue [hunk 1]
```diff
@@ -24,10 +24,11 @@ lineage: seqs: [
...
 				// Description of dashboard.
 				description?: string
-
-				// Version of the current dashboard data
-				revision: int64 | *-1 @grafanamaturity(NeedsExpertReview)
-
+				// This property should only be used in dashboards defined by plugins.  It is a quick check
+				// to see if the version has changed since the last time.  Unclear why using the version property 
+				// is insufficient.  
+				revision?: int64 @grafanamaturity(NeedsExpertReview)
```
Comment (nmarrs) [inline kinds/dashboard/dashboard_kind.cue:30]: Also wondering why this property is needed given we have a version property for dashboards.

It almost would make more sense for there to be a "isPlugin" bool vs another separate version tracking property 🤔

Comment (ryantxu) [reply kinds/dashboard/dashboard_kind.cue:30]: ya -- many other options, but this schema is really here to reflect the reality we have today

Comment (sdboyer) [reply kinds/dashboard/dashboard_kind.cue:30]: definitely some smashing together of concerns here

**Pull request merged**